plugins/destination/s3/client/spec.go

```diff
@@ -12,15 +12,19 @@ const (
 	FormatTypeJSON = "json"
 )
 
-type Spec struct {
-	Bucket         string     `json:"bucket,omitempty"`
-	Path           string     `json:"path,omitempty"`
+type FileSpec struct {
 	Format         FormatType `json:"format,omitempty"`
 	IncludeHeaders bool       `json:"include_headers,omitempty"`
 	Delimiter      rune       `json:"delimiter,omitempty"`
 	NoRotate       bool       `json:"no_rotate,omitempty"`
 }
 
+type Spec struct {
+	Bucket string `json:"bucket,omitempty"`
+	Path   string `json:"path,omitempty"`
+	FileSpec
+}
+
```

Comment:
@hermanschaaf @yevgenypats Is this what you were imagining?

In a future PR, we can move it to the filetypes package

Reply:
(Obviously this will require additional docs)